pyalgotrading/algobulls/connection.py

```diff
@@ -271,20 +279,112 @@ def get_report(self, strategy_code, trading_type, report_type, render_as_datafra
+                report: format and content of the report
+                html_dump: save it as a html file
+                pnl_df: dataframe containing pnl reports
+                initial_funds: initial funds to before starting the job
+            Returns:
+                Report details
+        """
+
+        order_report = None
+        # get pnl data and cleanup as per quantstats format
+        _returns_df = pnl_df[['entry_timestamp', 'pnl_absolute']]
+        _returns_df = _returns_df.set_index('entry_timestamp')
+        _returns_df["total_funds"] = _returns_df.pnl_absolute.cumsum() + initial_funds
+        _returns_df = _returns_df.dropna()
+
+        # Note: Quantstats has a potential bug. It cannot work with multiple entries having the same timestamp. For now, we are dropping multiple entries with the same entry_timestamp (else the quantstats code below would throw an error)
+        # Suggestion for workaround: For entries with same entry timestamps, we can slightly modify the entry timestamps by adding single-digit microseconds to make them unique
+        _returns_df = _returns_df[~_returns_df.index.duplicated(keep='first')]
+
+        # Extract the final column; note: timestamp is the index so that is available too
+        total_funds_series = _returns_df.total_funds
+
+        # select report type
+        if report == "metrics":
+            order_report = qs.reports.metrics(total_funds_series)
+        elif report == "full":
+            order_report = qs.reports.full(total_funds_series)
+
+        # save as html file
+        if html_dump:
+            all_strategies = self.get_all_strategies()
+            strategy_name = all_strategies.loc[all_strategies['strategyCode'] == strategy_code]['strategyName'].iloc[0]
+            qs.reports.html(total_funds_series, title=strategy_name, output='', download_filename=f'report_{strategy_name}_{time.time():.0f}.html')
+
+        return order_report
+
+    def start_job(self, strategy=None, start=None, end=None, instruments=None, lots=1, parameters=None, candle=None, mode=StrategyMode.INTRADAY, initial_funds_virtual=1e9, delete_previous_trades=True, trading_type=None, broker_id=None,
+                  broking_details=None, **kwargs):
+        """
+        Submit a BT/PT/RT job for a strategy on the AlgoBulls Platform
 
         Args:
             strategy: Strategy code
-            start: Start date/time
-            end: End date/time
+            start: Start date-time/time
+            end: End date-time/time
             instruments: Instrument key
             lots: Number of lots of the passed instrument to trade on
             parameters: Parameters
             candle: Candle interval
             mode: Intraday or delivery
             delete_previous_trades: Delete data for previous trades
+            initial_funds_virtual: virtual funds allotted before the backtesting starts
+            trading_type: type of trading : PT/BT/RT
+            broker_id: ID of the broker
```

Comment:
`broker_id` parameter not required as everything is coming through `broking_details`. Please clean everywhere.

Reply:
I think broker ID is required for RT